packages/amplify-migration-e2e/src/templategen.ts

```diff
@@ -0,0 +1,175 @@
+  cwd = cwd ?? process.cwd();
+  const processResult = execa.sync(command, {
+    cwd,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+    encoding: 'utf-8',
+    shell: true,
+  });
+
+  if (processResult.exitCode === 0) {
+    return processResult.stdout;
+  } else {
+    throw new Error(`Command exit code: ${processResult.exitCode}, message: ${processResult.stderr}`);
+  }
+}
+
+async function executeCreateStackRefactorCallCommand(command: string, cwd: string) {
+  const processResult = JSON.parse(await executeCommand(command, cwd));
+  const stackRefactorId = processResult.StackRefactorId;
+  return stackRefactorId;
+}
+
+async function assertStepCompletion(command: string) {
+  const processResult = JSON.parse(await executeCommand(command));
+  return processResult.Stacks[0].StackStatus;
+}
+
+async function assertRefactorStepCompletion(command: string) {
+  const processResult = JSON.parse(await executeCommand(command));
+  return processResult.Status;
+}
+
+async function executeStep1(cwd: string, commands: string[]) {
+  await executeCommand(commands[0], cwd);
+  await retry(
+    () => assertStepCompletion(commands[1]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+}
+
+async function executeStep2(cwd: string, commands: string[]) {
+  await executeCommand(commands[0], cwd);
+  await retry(
+    () => assertStepCompletion(commands[1]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+}
+
+async function executeStep3(cwd: string, commands: string[], bucketName: string) {
+  setEnvVariable('BUCKET_NAME', bucketName);
+  await executeCommand(commands[1], cwd);
+  await executeCommand(commands[2], cwd);
+  const stackRefactorId = await executeCreateStackRefactorCallCommand(commands[3], cwd);
+  setEnvVariable('STACK_REFACTOR_ID', stackRefactorId);
+  await retry(
+    () => assertRefactorStepCompletion(commands[5]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+  await executeCommand(commands[6], cwd);
+  await retry(
+    () => assertRefactorStepCompletion(commands[7]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+}
+
+export async function stackRefactor(projRoot: string, category: string, bucketName: string) {
```

Comment:
nit: make `category` param a type to constrain the possibilities:  `export RefactorCategory = 'auth' | 'storage'`